autorally_core/src/autorally_chassis/AutoRallyChassis.cpp

```diff
@@ -0,0 +1,583 @@
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+/**********************************************
+ * @file AutoRallyChassis.cpp
+ * @author Brian Goldfain <bgoldfai@gmail.com>
+ * @date July 10, 2016
+ * @copyright 2016 Georgia Institute of Technology
+ * @brief Interface for an AutoRally chassis
+ *
+ * @details Implementation of the AutoRallyChassis class that controls and receives state information from an AutoRally
+ *          chassis
+ ***********************************************/
+
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string.hpp>
+
+#include <pluginlib/class_list_macros.h>
+
+#include "AutoRallyChassis.h"
+
+PLUGINLIB_DECLARE_CLASS(autorally_core, AutoRallyChassis, autorally_core::AutoRallyChassis, nodelet::Nodelet)
+
+namespace autorally_core
+{
+
+AutoRallyChassis::~AutoRallyChassis()
+{
+  chassisControlTimer_.stop();
+}
+
+void AutoRallyChassis::onInit()
+{
+  ros::NodeHandle nh = getNodeHandle();
+  ros::NodeHandle nhPvt = getPrivateNodeHandle();
+  std::string port;
+  if(!nh.getParam(getName()+"/port", port))
+  {
+    NODELET_ERROR("AutoRallyChassis: could not get chassis interface port");
+  }
+
+  loadChassisConfig();
+  loadChassisCommandPriorities();
+  serialPort_.init(nh, getName(), "", "AutoRallyChassis", port, true);
+  
+  double commandRate = 0.0;
+  double chassisCommandMaxAge = 0.0;
+  double runstopMaxAge = 0.0;
+
+  if(!nhPvt.getParam("commandRate", commandRate) ||
+     !nhPvt.getParam("commandMaxAge", chassisCommandMaxAge) ||
+     !nhPvt.getParam("runstopMaxAge", runstopMaxAge))
+  {
+    NODELET_ERROR_STREAM(getName() << " could not get all startup params");
+  }
+  chassisCommandMaxAge_ = ros::Duration(chassisCommandMaxAge);
+  runstopMaxAge_ = ros::Duration(runstopMaxAge);
+
+  chassisStatePub_ = nh.advertise<autorally_msgs::chassisState>
+                     ("chassisState", 1);
+  wheelSpeedsPub_ = nh.advertise<autorally_msgs::wheelSpeeds>
+                     ("wheelSpeeds", 1);
+  chassisCommandPub_ = nh.advertise<autorally_msgs::chassisCommand>
+                     ("RC/chassisCommand", 1);
+
+  for (auto& mapIt : chassisCommands_)
+  {
+    std::string topic = mapIt.first+"/chassisCommand";
+    ros::Subscriber sub = nh.subscribe(topic, 1,
```

Comment:
This information is sent as ROS_INFO messages when the data is read from the paramter server on startup